src/nspawn/nspawn.c

```diff
@@ -3783,6 +3841,71 @@ static int run(int master,
         return 1; /* loop again */
 }
 
+static int initialize_rlimits(void) {
+
+        /* The default resource limits the kernel passes to PID 1, as per kernel 4.16. Let's pass our container payload
+         * the same values as the kernel originally passed to PID 1, in order to minimize differences between host and
+         * container execution environments. */
```

Comment:
Yeah, this seems fishy. I don't see why we should raise limits above that of a normal unit.